### DISCUSSION TITLE
🛤️ Which path should we follow? | Query Rendering

### DISCUSSION THREAD

**Discussion by Rooyca** [Rooyca/query-json#2] [Polls]
I am currently at a crossroads and I would like your opinion.
---

Right now, when we do a query, the result replaces the query and there is no way to edit the query (unless we press `Ctrl+Z`). But this way, the value is actually in the document. My question is, should this continue to be the way it is or should we change to a format where the query can be edited, i.e. the result is only rendered in the document. A disadvantage of this method is that the query must be run more than once (in fact, it must be run many times more than with the current method).

**Comment by huwasm** [comment]
i am to new in coding to say any, sorry but cool

**Reply by Rooyca** [reply]
That's OK. I hope you find this plugin useful.

**Comment by garbsclassic** [comment]
I've been looking for a way to do exactly what you're proposing, for JSON data that updates semi-regularly (not w/in the same session).

To avoid multiple query runs, can the result be cached and only shown in the render view, while the query remains in source mode? Not sure if that's feasible or too potentially confusing.
